rfcs/0068-error-record-table.md

```diff
@@ -0,0 +1,85 @@
+---
+feature: error_record_table
+authors:
+  - "TennyZhuang"
+start_date: "2023/07/31"
+---
+
+# Error Record Table
+
+## Summary
+
+Our current streaming engine does not help users to discover, debug, and handle errors well. When user met an data record error, they can only find a log record like ``ExprError: Parse error: expected `,` or `]` at line 1 column 10 (ProjectExecutor: fragment_id=19007)``.
```

Comment:
FWIW, currently we also have error reporting to Prometheus (https://github.com/risingwavelabs/risingwave/issues/7824), but it do not contains original data (seemly for the same reason described in the Alternatives section)

Reply:
@neverchanje  proposed similar ideas in https://github.com/risingwavelabs/risingwave/issues/7803

> But when it comes to production deployment, we will need to provide an option that allows users to persist lost messages somewhere (maybe in Hummock or a log store) so that they can search for what reason and what data were lost (maybe through ElasticSearch).

Closed as "we are now able to find the errors from Grafana", but we haven't supported "allow users to persist lost messages" yet.